docs/architecture/adr-059-evidence-composition-and-lifecycle.md

```diff
@@ -0,0 +1,256 @@
+
+  - If lunatic the validators from the common val set that signed in the conflicting block
+
+  - If amnesia none (we can't know which validators are malicious). This also means that we don't
+  send amnesia evidence to the application
+
+- For each validator, check the look up table to make sure there already isn't evidence against this validator
+
+
+After verification we persist the evidence with the following key `height/hash` to the pending evidence database in the evidence pool with the following format
+
+```go
+type EvidenceInfo struct {
+  ev Evidence
+  time time.Time
+  validators []Validator
+  totalVotingPower int64
+}
+```
+
+We persist more information than just the evidence as this will be useful in forming the ABCI Evidence that we will come to later
+
+Receiving broadcasts from other evidence reactors works in the same manner.
+
+When it comes to prevoting and precomitting a proposal that contains evidence, the full node will once again
+call upon the evidence pool to verify the evidence using `CheckEvidence(ev []Evidence)`:
+
+This loops through all the evidence first checking that nothing has been duplicated then:
+
+runs `fastCheck(ev evidence)` which works similar to `Has` but with `LightClientAttackEvidence` if it has the
+same hash it then checks to see that the validators it has are all signers in the commit of the conflicting header. If it doesn't pass fast check (because it hasn't seen the evidence before) then it will have to verify the evidence.
+
+runs `Verify(ev Evidence)` - Note: this also saves the evidence to the db as mentioned before.
+
+
+The final part of the lifecycle is when the block is committed and the `BlockExecutor` tries to apply state. Instead of trying to form the information necessary to produce the ABCI evidence it gets the evpool to create the evidence in the update step as follows.
+
+`Update(Block, State) []abci.Evidence`
+
+```go
+abciResponses.BeginBlock.ByzantineValidators = evpool.Update(block, state)
+```
+
+Update does the following:
+- increments state which keeps track of both the current time and height used for measuring expiry
+- marks evidence as committed and saves to db. This prevents validators from proposing committed evidence in the future
+  Note: the db just saves the height and the hash. There is no need to save the entire committed evidence
+- forms abci evidence:
+```go
+for _, val := range evInfo.Validators {
+  abciEv = append(abciEv, &abci.Evidence{
+    Type: evType,
+    Validator: val,
+    Height: evInfo.ev.Height(),
+    Time: evInfo.time,
+    TotalVotingPower: evInfo.totalVotingPower
+  })
+}
+```
+and compiles it all together (note for `DuplicateVoteEvidence` the validators array size is 1)
+- removes expired evidence from both pending and committed
+
+The abci evidence is then sent via the `BlockExecutor` to the application.
+
+To summarize, we can see the lifecycle of evidence as such:
+
+![evidence_lifecycle](../imgs/evidence_lifecycle.png)
```

Comment:
This is a really great diagram! Did you make this?? (And how?) 

I especially love the nod to transit maps and the little train at the top. 

A few notes:

1. The "key" for the different types looks like it's part of the flow--pulling it a little farther away would make it easier to parse. 
2. Similarly, a little star that says "start here, when the light client notices a divergence" would be helpful. Or maybe two different stars: "start here for light client attacks" and "start here for duplicate votes"
3. What do the big white (empty) rectangles represent? 
4. Is there significance to the ordering of things inside the evidence pool?

Reply:
3. They are just stations/platforms
4. Not really, but I tried to capture the flow of information